### ISSUE TITLE
Model and datasets

### ISSUE THREAD

**Issue by PawelGawlikDev** [truekendor/chessboard-image-to-fen#5]
Hi, This project is great! I'm interest how did you train model. Can you share code source code used to create nn used in this project?

**Comment by truekendor** [comment]
@PawelGawlikDev Hi! Thanks for the kind words, I'm glad you liked the project! 

I'm happy to share the code I used to train the neural network. However, some parts of the training data include personal content ( parts of my YouTube feed, wallpapers with pictures, etc.), so I won't be able to include the full dataset tied to that private data (mostly chessboard detection dataset, not the piece classification one)

That said, I’ll clean things up and share the training code and general structure soon. Let me know if you're looking for anything specific in the meantime!

**Comment by truekendor** [comment]
Dataset for piece classification is pretty clean and easy to reproduce, so I can share it no problem

**Comment by PawelGawlikDev** [comment]
@truekendor awesome! I'd like to see how this model works and try to add some improvements to NN if it's possible.

**Comment by PawelGawlikDev** [comment]
Sorry for miss close. If it's not problem for you it can be close

**Comment by truekendor** [comment]
@PawelGawlikDev it's all good, don't worry

**Comment by PawelGawlikDev** [comment]
@truekendor any update?

**Comment by truekendor** [comment]
@PawelGawlikDev Just finished cleaning up detection dataset
I'll make sure training script still works and tag you :)

**Comment by PawelGawlikDev** [comment]
Awesome, thanks

**Comment by truekendor** [comment]
@PawelGawlikDev Pieces classification script's still not ready cause i lost it somewhere, but dataset can be send today along with detection one

**Comment by PawelGawlikDev** [comment]
Ok, good for start :)

**Comment by PawelGawlikDev** [comment]
@truekendor If you have problem with pieces classification don't worry. I think I'll be able to deal with it by myself or just get it later. For now more important for me is detecting chessboards and datasets.

**Comment by truekendor** [comment]
@PawelGawlikDev Hey. Yeah, I'm busy debugging rn, for some reason my old script not working as expected.
I mean the detection one, unfortunately

Model can be trained no problem, but not exported to TF.js :(  .
This was the main problem last time I've tried to train it aaand now it's back

I can send you current script and dataset, but it wont work 100% as expected

**Comment by truekendor** [comment]
@PawelGawlikDev Anyways

Dataset for detection
https://drive.google.com/drive/folders/1hUGv62uOGyjLLcgBcNxKOTOApTWB6T76?usp=sharing

Google colab link
https://colab.research.google.com/drive/1VKjnalfFDmrrhU15dwV3BiMeOBwKa9LY?usp=sharing

**Comment by PawelGawlikDev** [comment]
@truekendor Awesome. If something not working with classifier you can send just structure of code. Maybe I'll be able to fix it.

**Comment by PawelGawlikDev** [comment]
@truekendor hi, quick followup. Can you share code of classification model? Dataset is not needed.

**Comment by truekendor** [comment]
@PawelGawlikDev Classification only, no dataset? This might be a bit easier, as long as you don't mind training model with nodeJS like i did :) 

This is not recommended way of training the model, but it will work

**Comment by PawelGawlikDev** [comment]
@truekendor Classification code can be helpful for me. I'm trying to make it based on model.jeon file but I have some problem so I'd like to see how you did it. Also I did my prototype in NodeJs because of some problem with convert model from python.

**Comment by PawelGawlikDev** [comment]
Also I'm training it on many images of single board tiles with pieces or empty of your datase looks different format tell me how example data should looks like

**Comment by truekendor** [comment]
@PawelGawlikDev  That makes sense now. Can I suggest a YT video that pretty much summarizes what I did?
This video in particular [LINK](https://youtu.be/PN4asCDITNg) and the whole course was my guideline on how to do it. Hope this will be helpful for you too!

If this won't be enough guess I'll try to rewrite this from scratch and share the code with you in a new repo

**Comment by PawelGawlikDev** [comment]
@truekendor Rewrite from scratch can by very helpfull with found what I can fix in my code. If it will help you I can share code that I have for now and you can tell me what you did different then I.

**Comment by truekendor** [comment]
Ok then, rewrite it is! I would ask you to wait until the middle of next week, since I have other work to do.

**Comment by truekendor** [comment]
Not sure if I'm the right person to review someone's code, most of the things i've done in this project are just minor modifications of other peoples courses.

Now I understand that resources used to do this project can be helpful for someone else. I'll provide links to resources that at all allowed this project to exist later with the classification code

**Comment by truekendor** [comment]
@PawelGawlikDev Hey, just a quick update!
I'm still working on the code, can't quite get it to work the way it was (npm package issue), but all the main stuff is almost ready
The problem is i can't test if it works or not because of a said package issues

**Comment by PawelGawlikDev** [comment]
Good luck and I'm keeping my fingers crossed

**Comment by PawelGawlikDev** [comment]
@truekendor  Hi, I'd like to ask how working with npm package issue is going on?

**Comment by truekendor** [comment]
@PawelGawlikDev Hey! Yes, I think all's ready, i'll publish it later today. The package problem is quite serious - the node-tensorflow package is now not available. This means that feature extraction part is sooo slow

**Comment by truekendor** [comment]
@PawelGawlikDev I tried to find some workaround but to no avail

**Comment by PawelGawlikDev** [comment]
@truekendor Ok, so I'm waiting publishing code.

**Comment by truekendor** [comment]
@PawelGawlikDev  [link](https://github.com/truekendor/node-chess-classifier)

There's a quick classification setup
Download node.js and npm, then run `npm install` in projects directory

The idea behind this is pretty simple. First we extract tile features from dataset with mobilenet, then we run train command to fit extracted features input to one-hot encoded piece type output 

Default training settings is for 5 epochs only, but you could change it, 5 is just a mock of a real process. Just to make sure that it works

**Comment by PawelGawlikDev** [comment]
@truekendor One last question do you remember how many epochs was in original code? I'm trying to get model that will work the same as in chessboard-image-to-fen project

**Comment by truekendor** [comment]
@PawelGawlikDev About 300-350, but iirc the target was 90%+ accuracy on my (now lost) dataset
